src/java/com/phenix/pct/PCTWSBroker.java

```diff
@@ -0,0 +1,450 @@
+
+    /**
+     * Server log file append. Default value is true.
+     * 
+     * @param append Boolean
+     */
+    public void setServerLogFileAppend(boolean append) {
+        this.serverLogFileAppend = append;
+    }
+
+    /**
+     * Working directory of servers. No default value.
+     * 
+     * @param workDir File
+     */
+    public void setWorkDir(File workDir) {
+        this.workDir = workDir;
+    }
+
+    /**
+     * Controlling name server. Default value is NS1
+     * 
+     * @param ns String
+     */
+    public void setNameServer(String ns) {
+        this.nameServer = ns;
+    }
+
+    /**
+     * Server settings
+     * 
+     * @param sp ServerProcess (customized version of PCTRun)
+     */
+    public void addConfiguredServer(WSServerProcess sp) {
+        if (this.server == null) {
+            this.server = sp;
+        } else {
+            throw new BuildException("Only one server process...");
+        }
+    }
+
+    /**
+     * Do the work
+     * 
+     * @throws BuildException Something went wrong
+     */
+    @Override
+    public void execute() {
+        File propFile = null;
+
+        try {
+            this.checkAttributes();
+            // Choosing right properties file
+            if (this.file == null) {
+                propFile = new File(this.getDlcHome(), "properties/" + UBROKER_PROPERTIES);
+            } else {
+                propFile = getProject().resolveFile(this.file);
+            }
+            if (!propFile.exists())
+                throw new BuildException("Unable to find properties file "
+                        + propFile.getAbsolutePath());
+            // And do the work
+            writeDeltaFile();
+            Task task = getCmdLineMergeTask(propFile, tmpFile);
+            task.execute();
+        } catch (BuildException be) {
+            throw be;
+        } finally {
+            // Unwanted side effect of using PCTRun in server attribute, empty temporary files need
+            // to be cleaned
+            if (this.server != null)
+                this.server.cleanup();
+            this.cleanup();
+        }
+    }
+
+    /**
+     * Write files to be used as a parameter for the MergeProp shell script.
+     * 
+     * @throws BuildException Something went wrong during write.
+     * 
+     */
+    private void writeDeltaFile() {
+        try (Writer w = new FileWriter(tmpFile); PrintWriter bw = new PrintWriter(w)) {
+            bw.println("[UBroker.WS." + this.name + "]");
+            if (!"delete".equalsIgnoreCase(this.action)) {
+                bw.println("appserviceNameList=" + this.name);
+                bw.println("registerNameServer=1");
+                bw.println("controllingNameServer=" + this.nameServer);
+                bw.println("registrationMode=Register-IP");
+                bw.println("autoStart=" + (this.autoStart ? "1" : "0"));
+                // TODO Erm, this is crap... I should use something else to handle correctly quotes in command line
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)